### PR TITLE
Migrate slack bot

### PR DESCRIPTION
**Pull request by violetbrina**
- Move the gcp cost reporting bot to the cpg infrastructure.
- There are also one or two fixes to general Exception catches that I've fixed since adding ruff checking to the linter.
- Also a matching PR in [infra-private](https://github.com/populationgenomics/cpg-infrastructure-private/pull/364)
- Added Michael's suggestions for improving the flagging

Preview:
Screenshot 2024-07-23 at 5 31 41 PM

### CODE REVIEW

File: cpg_infra/billing_aggregator/driver.py [hunk 1]
```diff
@@ -181,21 +262,30 @@ def setup_aggregator_functions(self):
...
+
             # Create the function, the trigger and subscription.
             fxn, _ = self.create_cloud_function(
                 resource_name=f'billing-aggregator-{function}-billing-function',
-                name=function,
+                name=f'Aggregator {function.capitalize()}',
                 source_file=f'{function}.py',
                 service_account=self.config.billing.coordinator_machine_account,
-                source_archive_object=source_archive_object,
+                soure_bucket=self.source_bucket.name,
```
Comment (dancoates) [inline cpg_infra/billing_aggregator/driver.py:279]: misspelled param `soure_bucket`

**Review by dancoates — CHANGES_REQUESTED** [review]
looks good, just a few little things

File: cpg_infra/billing_aggregator/driver.py [hunk 2]
```diff
@@ -181,21 +262,30 @@ def setup_aggregator_functions(self):
             if function == 'seqr':
                 # 2GB is not enough for seqr
                 memory = '2560M'
+
+            # Create the source archive
+            source_archive = self.create_source_archive(
+                'billing-aggregator-source-code',
```
Comment (milo-hyben) [inline cpg_infra/billing_aggregator/driver.py:268]: Wonder if this should be `f'billing-aggregator-{function}-source-code'`

**Review by milo-hyben — CHANGES_REQUESTED** [review]
Just a small thing from my end to add to already mentioned by Dan.
Otherwise looking very good.

File: cpg_infra/billing_aggregator/driver.py (hunk 2 above)
Comment (violetbrina) [reply cpg_infra/billing_aggregator/driver.py:268]: Ah good call. I don't think so, it get's used by all the aggregator functions. But I have moved it out of the function loop for clarity

File: cpg_infra/billing_aggregator/driver.py (hunk 1 above)
Comment (violetbrina) [reply cpg_infra/billing_aggregator/driver.py:279]: Fixed

File: cpg_infra/billing_aggregator/driver.py [hunk 3]
```diff
@@ -139,6 +133,109 @@ def slack_channel(self):
...
+        `cost-report` topic, e.g. using a daily schedule like `0 9 * * *`.
+        The payload can be abitrary, as it is ignored in the Cloud Function.
+        1. Install the Cloud Function that gets triggered when a message to the
+        `cost-report` Pub/Sub topic is posted. Set `$QUERY_TIME_ZONE` to your local
+        time zone, e.g. `Australia/Sydney`.
+        """
+
+        # Create source archive
+        source_archive = self.create_source_archive(
+            'billing-aggregator-source-code',
```
[the remark this answers is not in the document]
Comment (violetbrina) [reply cpg_infra/billing_aggregator/driver.py:183]: Done

File: cpg_infra/billing_aggregator/driver.py [hunk 4]
```diff
@@ -139,6 +133,109 @@ def slack_channel(self):
...
+        bigquery_billing_table = self.config.billing.gcp.source_bq_table
+        billing_account_id = self.config.billing.gcp.account_id
+        region = self.config.gcp.region
+
+        time_zone = self.config.billing.gcp_cost_reporting.timezone
+        service_account = self.config.billing.gcp_cost_reporting.machine_account
+        slack_channel = self.config.billing.gcp_cost_reporting.slack_channel
+
+        """
+        1. Create a new Pub/Sub topic in the `billing-admin-290403` project, named
```
[the remark this answers is not in the document]
Comment (violetbrina) [reply cpg_infra/billing_aggregator/driver.py:171]: Comment removed!

**Review by milo-hyben — APPROVED** [review]
Changes looking good to me!

**Review by dancoates — APPROVED** [review]
changes look good, thanks!

**Pull request merged**